### PR TITLE
Downgrade LIMITED_API_VERSION, if python3 is older than 3.6

### PR DESCRIPTION
**Pull request by asmorkalov**
Fixes https://github.com/opencv/opencv/pull/24773#discussion_r1450650158

```
force_builders=Win64 OpenCL
```

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: cmake/OpenCVDetectPython.cmake [hunk 1]
```diff
@@ -295,7 +295,10 @@ find_python("${OPENCV_PYTHON3_VERSION}" "${MIN_VER_PYTHON3}" PYTHON3_LIBRARY PYT
 OCV_OPTION(PYTHON3_LIMITED_API "Build with Python Limited API (not available with numpy >=1.15 <1.17)" NO
            VISIBLE_IF PYTHON3_NUMPY_VERSION VERSION_LESS "1.15" OR NOT PYTHON3_NUMPY_VERSION VERSION_LESS "1.17")
 if(PYTHON3_LIMITED_API)
-  set(PYTHON3_LIMITED_API_VERSION "0x03060000" CACHE STRING "Minimal Python version for Limited API")
+  # set LIMITED API version that corresponds to current python version, if python3 is too old
+  set(PYTHON3_LIMITED_API_VERSION
+      IF ${PYTHON}_VERSION_STRING VERSION_GREATER "3.5.999" THEN "0x03060000" ELSE "0x030${PYTHON3_VERSION_MINOR}0000"
```
Comment (opencv-alalek) [inline cmake/OpenCVDetectPython.cmake:300]: Which CMake version supports conditionals in `set()`?

File: cmake/OpenCVDetectPython.cmake [hunk 2]
```diff
@@ -295,7 +295,12 @@ find_python("${OPENCV_PYTHON3_VERSION}" "${MIN_VER_PYTHON3}" PYTHON3_LIBRARY PYT
 OCV_OPTION(PYTHON3_LIMITED_API "Build with Python Limited API (not available with numpy >=1.15 <1.17)" NO
            VISIBLE_IF PYTHON3_NUMPY_VERSION VERSION_LESS "1.15" OR NOT PYTHON3_NUMPY_VERSION VERSION_LESS "1.17")
 if(PYTHON3_LIMITED_API)
-  set(PYTHON3_LIMITED_API_VERSION "0x03060000" CACHE STRING "Minimal Python version for Limited API")
+  # set LIMITED API version that corresponds to current python version, if python3 is too old
+  if(${PYTHON}_VERSION_STRING VERSION_GREATER "3.5.999")
```
Comment (mshabunin) [inline cmake/OpenCVDetectPython.cmake:299]: Perhaps it would be more readable to use negative condition (+avoid docstring duplication):
```.cmake
set(_default_ver "0x03060000")
if(${PYTHON}_VERSION_STRING VERSION_LESS "3.6")
  # fix for older pythons
  set(_default_ver "0x030${PYTHON3_VERSION_MINOR}0000")
endif()
set(PYTHON3_LIMITED_API_VERSION ${_default_ver} CACHE STRING "Minimal Python version for Limited API")
```

**Review by mshabunin — APPROVED** [review]
Overall looks good to me, except cosmetic comments. 
Verified with Ubuntu 16:
```
--   Python 3:
--     Interpreter:                 /usr/bin/python3 (ver 3.5.2)
--     Libraries:                   /usr/lib/x86_64-linux-gnu/libpython3.5m.so (ver 3.5.2)
--     Limited API:                 YES (ver 0x03050000)
--     numpy:                       /usr/lib/python3/dist-packages/numpy/core/include (ver 1.11.0)
--     install path:                lib/python3.5/site-packages/cv2/python-3
```

File: cmake/OpenCVDetectPython.cmake [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Find specified Python version
+ #Find specified Python version
```
Comment (opencv-alalek) [inline cmake/OpenCVDetectPython.cmake:1]: unrelated change

**Comment by mshabunin** [comment]
It seems that you've reverted previous cosmetic fix while fixing unrelated change, was it intended?

**Comment by asmorkalov** [comment]
@mshabunin @opencv-alalek I fixed the issue, please take a look.

**Review by opencv-alalek — APPROVED** [review]
